### PR TITLE
More netplay improvements

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

**Comment by turol** [comment]
What about #369?

**Comment by Vagabond** [comment]
> What about #369?

I investigated using it, but portability is a mess. I ended up reimplementing most of the core ideas. We could probably integrate the ggpo library now if we wanted to, because we should have the 'save state' that GGPO requires now, because of the work in #565

**Comment by Vagabond** [comment]
> > What about #369?
> 
> I investigated using it, but portability is a mess. I ended up reimplementing most of the core ideas. We could probably integrate the ggpo library now if we wanted to, because we should have the 'save state' that GGPO requires now, because of the work in #565

there's some forks of ggpo that worked on linux, so it's not unreasonable to do, at this point.

**Review by mrannanj — CHANGES_REQUESTED** [review]

File: src/game/utils/settings.c [hunk 1]
```diff
@@ -171,7 +172,11 @@ void settings_load_fields(void *st, const field *fields, int nfields) {
                 // make a copy of the string
                 char **s = fieldstr(st, f->offset);
                 omf_free(*s);
-                *s = strdup(conf_string(f->name));
+                if(conf_string(f->name)) {
+                    *s = strdup(conf_string(f->name));
+                } else {
+                    *s = NULL;
```
[the remark this answers is not in the document]
Comment (Vagabond) [reply src/game/utils/settings.c]: Fixed

File: src/controller/net_controller.c [hunk 2]
```diff
@@ -177,19 +195,25 @@ void rewind_and_replay(wtf *data, game_state *gs_current) {
...
+        for(int dynamic_wait = ticks; dynamic_wait > 0; dynamic_wait--) {
             // Tick scene
             game_state_dynamic_tick(gs, true);
         }
 
         // feed in the inputs
         // XXX this is a hack for now
 
+        if((ev->events[0] || ev->events[1]) && ev->tick <= data->last_received_tick) {
+            int sz = snprintf(buf, 254, "%d - player 1 %d -- player 2 %d\n", ev->tick, ev->events[0], ev->events[1]);
```
[the remark this answers is not in the document]
Comment (Vagabond) [reply src/controller/net_controller.c]: Fixed

File: src/game/utils/settings.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrannanj) [reply src/game/utils/settings.c]: But is this the correct change overall, can `conf_string(f->name)` return NULL here?

**Review by mrannanj — APPROVED** [review]

**Pull request merged**